src/coprocessor/dag/expr/ctx.rs

```diff
@@ -172,15 +157,9 @@ impl EvalConfig {
         }
     }
 
-    pub fn set_by_flags(&mut self, flags: u64) -> &mut Self {
-        self.set_ignore_truncate((flags & FLAG_IGNORE_TRUNCATE) > 0)
-            .set_truncate_as_warning((flags & FLAG_TRUNCATE_AS_WARNING) > 0)
-            .set_overflow_as_warning((flags & FLAG_OVERFLOW_AS_WARNING) > 0)
-            .set_in_insert_stmt((flags & FLAG_IN_INSERT_STMT) > 0)
-            .set_in_update_or_delete_stmt((flags & FLAG_IN_UPDATE_OR_DELETE_STMT) > 0)
-            .set_in_select_stmt((flags & FLAG_IN_SELECT_STMT) > 0)
-            .set_pad_char_to_full_length((flags & FLAG_PAD_CHAR_TO_FULL_LENGTH) > 0)
-            .set_divided_by_zero_as_warning((flags & FLAG_DIVIDED_BY_ZERO_AS_WARNING) > 0)
+    pub fn set_by_flags(&mut self, new_value: Flags) -> &mut Self {
```

Comment:
The behaviour is override instead of mask now. You may change the function name to set_flag.

Reply:
ok, I will rewrite the code